bin/packages/build.js

```diff
@@ -29,41 +30,77 @@ function getPackageName( file ) {
 }
 
 /**
- * Returns a stream transform which maps an individual stylesheet to its
- * package entrypoint. Unlike JavaScript which uses an external bundler to
- * efficiently manage rebuilds by entrypoints, stylesheets are rebuilt fresh
- * in their entirety from the build script.
+ * Helper higher order function for creating stream transforms that map from
+ * one input file to none, one, or multiple output files.
+ *
+ * @param {Function} mappingFunc An async mapping function that receives a file
+ * 								 path and returns either a falsy to indicate
+ * 								 the file should not be built, a string file
+ * 								 path to indicate that this single file should
+ * 								 be built, or an array of strings to indicate
+ * 								 multiple files should be built.
  *
  * @return {Transform} Stream transform instance.
  */
-function createStyleEntryTransform() {
-	const packages = new Set;
+function createEntryTransform( mappingFunc ) {
+	// Track any already built files to avoid duplication.
+	const fileSet = new Set;
 
 	return new Transform( {
 		objectMode: true,
 		async transform( file, encoding, callback ) {
-			// Only stylesheets are subject to this transform.
-			if ( path.extname( file ) !== '.scss' ) {
-				this.push( file );
+			const mapped = await mappingFunc( file );
+
+			// Ignore this file when a falsey is provided.
+			if ( ! mapped ) {
 				callback();
 				return;
 			}
 
-			// Only operate once per package, assuming entries are common.
-			const packageName = getPackageName( file );
-			if ( packages.has( packageName ) ) {
+			if ( isString( mapped ) && ! fileSet.has( mapped ) ) {
+				// Handle a single path.
+				this.push( mapped );
+				fileSet.add( mapped );
 				callback();
 				return;
 			}
 
-			packages.add( packageName );
-			const entries = await glob( path.resolve( PACKAGES_DIR, packageName, 'src/*.scss' ) );
-			entries.forEach( ( entry ) => this.push( entry ) );
-			callback();
+			if ( Array.isArray( mapped ) ) {
+				// Handle an array of mapped paths.
+				mapped.forEach( ( filePath ) => {
+					if ( ! fileSet.has( filePath ) ) {
+						this.push( filePath );
+						fileSet.add( mapped );
+					}
+				} );
+				callback();
+			}
 		},
 	} );
 }
 
+/**
+ * Returns a stream transform which maps an individual stylesheet to its
+ * package entrypoint. Unlike JavaScript which uses an external bundler to
+ * efficiently manage rebuilds by entrypoints, stylesheets are rebuilt fresh
+ * in their entirety from the build script.
+ *
+ * @return {Transform} Stream transform instance.
+ */
+function createStyleEntryTransform() {
+	return createEntryTransform( async ( file ) => {
+		// Only stylesheets are subject to this transform.
+		if ( path.extname( file ) !== '.scss' ) {
+			return file;
+		}
+
+		// Build all root level stylesheets for the package.
+		const packageName = getPackageName( file );
+		const rootScssFiles = await glob( path.resolve( PACKAGES_DIR, packageName, 'src/*.scss' ) );
```

Comment:
The capitalization of this as an acronym should be `rootSCSSFiles`.

https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#abbreviations-and-acronyms

That said, I don't think we need a variable here, just `return await ...`